File: ruby/lib/warehouse_actions/warehouse_actions.rb
require_relative './stock_report_command'
require_relative './replenish_stock_command'

module WarehouseActions

	class WarehouseActions

		def initialize(warehouse, catalogue)
			@warehouse = warehouse
			@catalogue = catalogue
		end

		def stock_report
			StockReportCommand.new(@warehouse)
		end

		def replenish_stock
			ReplenishStockCommand.new(@warehouse, @catalogue)
		end

	end
end

File: python/warehouse_actions/warehouse_actions.py
from stock_report_command import StockReportCommand
from replenish_stock_command import ReplenishStockCommand

class WarehouseActions:

    def __init__(self, warehouse, catalogue):
        self.warehouse = warehouse
        self.catalogue = catalogue

    def stock_report(self):
        return StockReportCommand(self.warehouse)

    def replenish_stock(self):
        return ReplenishStockCommand(self.warehouse, self.catalogue)

